src/ds/strings.md

```diff
@@ -0,0 +1,109 @@
+Redis strings are used to store strings, numbers, images, serialized objects, and other kinds of information. They are the simplest of the Redis data types.
+
+As a first example, use the `SET` command to store the value "Deimos", a type of bike, at key "bike:1".
+
+```redis Create a new key
+SET bike:1 "Deimos"
+```
+
+You can retrieve the data using the `GET` command.
+
+```redis Retrieve the data
+GET bike:1
+```
+
+To see if a key exists, use the `EXISTS` command.
+
+```redis Do these keys exist?
+EXISTS bike:1
+EXISTS bike:2
+```
+
+The `EXISTS` command returns `1` when a key exists and `0` when it does not exist. The `bike:1` key exists, so Redis returns a value of `1`. The `bike:2` key does not exist, so Redis returns `0`.
+
+Use the `DEL` to delete a key and its data.
+
+```redis Delete bike:1
+DEL bike:1
+EXISTS bike:1
+```
+
+## Strings as counters
+
+If you store integer data in a string key, you can use these commands to increment and decrement its value:
+
+- `INCR` - increment a number stored in a key by `1`.
+- `INCRBY` - increment a number stored in a key by a specific positive or negative amount.
+- `DECR` - decrement a number stored in a key by `2`.
+- `DECRBY` - decrement a number stored in a key by a specific positive or negative amount.
+
+```redis INCR usage
+SET bikes:total_number 10
+INCR bikes:total_number
+INCR bikes:total_number
+```
+
+```redis Use INCR to set a key to 1
+DEL bikes:total_number
+INCR bikes:total_number
+```
+
+Note how the `INCR bikes:total_number` command that follows the `DEL` command re-creates the `bikes:total_number` key and sets its value to `1`.
+
+```redis INCRBY usage
+INCRBY bikes:total_number 100
+```
+
+```redis DECR and DECRBY usage
+DECR bikes:total_number
+DECRBY bikes:total_number 10
+```
+
+Using any of the `INCR*` or `DECR*` commands to modify the value of a key is an atomic operation. If two clients were to increment a key's value simultaneously, both operations would increment the key.
+
+All the Redis operations implemented by single commands are atomic, including those that operate on more complex data structures. So, when you use a command that modifies the value of a key, any concurrent operations will also succeed.
+
+## Key expiration
```

Comment:
Why is key expiration explained under strings? It refers to any key in Redis and should appear under a separate section.